waiter/src/waiter/scaling.clj

```diff
@@ -594,6 +593,40 @@
             (System/exit 1)))))
     {:exit exit-chan
      :query query-chan
-     :query-state-fn (fn query-state-fn [] @state-atom)
-     :query-service-state-fn (fn query-service-state-fn [query-params]
+     :query-state-fn (fn query-autoscaler-state-fn [] @state-atom)
+     :query-service-state-fn (fn query-autoscaler-service-state-fn [query-params]
                                (query-autoscaler-service-state @state-atom query-params))}))
+
+(defn start-scaling-mode-tracker
```

Comment:
Chose to create a separate daemon to maintain and populate the scaling mode, instead of adding the responsibility to `waiter.scaling/scale-services`